js/controller.js

```diff
@@ -831,6 +779,85 @@ OO.plugin("Html5Skin", function (OO, _, $, W) {
       this.renderSkin();
     },
 
+    /********************************************************************
+     MAIN VIDEO RELATED EVENTS
+     *********************************************************************/
+
+    //merge and load config data
+    loadConfigData: function(params, settings, data) {
+      if (data) {
+        SkinJSON = data;
+      }
+
+      //override data in skin config with possible inline data input by the user
+      $.extend(true, SkinJSON, params.skin.inline);
+      //override state settings with defaults from skin config and possible local storage settings
+      $.extend(true, this.state.closedCaptionOptions, SkinJSON.closedCaptionOptions, settings.closedCaptionOptions);
+      this.state.config = SkinJSON;
+
+      //load config language json if exist
+      if (SkinJSON.localization.availableLanguageFile) {
+        SkinJSON.localization.availableLanguageFile.forEach(function(languageObj){
+          if (languageObj.languageFile) {
+            $.getJSON(languageObj.languageFile, function(data) {
+              Localization.languageFiles[languageObj.language] = data;
```

Comment:
languageObj isn't modified it's read-only. Only the Localization object is updated if lang file is specified in config file.